instrumentation/mongo/mongo-4/javaagent/mongo-4-javaagent.gradle

```diff
@@ -0,0 +1,26 @@
+apply from: "$rootDir/gradle/instrumentation.gradle"
+
+muzzle {
+  pass {
+    group = "org.mongodb"
+    module = "mongodb-driver-core"
+    // this instrumentation is backwards compatible with early versions of the new API that shipped in 3.7
```

Comment:
I'm not sure if this complicates matters, but the upcoming 4.2.0 release will allow construction of a legacy `com.mongodb.MongoClient` with a `com.mongodb.MongoClientSettings`.  See https://github.com/mongodb/mongo-java-driver/commit/15b70cf7e5d41875bc46e57897f98d43ca60f4f6#diff-9edd83ced1b6f12502db95a96af13ed57e647efd84d2ddf5d040af4a653fb1edR211 (you have to expand the MongoClient diffs as they are large)